compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/cfa/FirPropertyInitializationAnalyzer.kt

```diff
@@ -66,10 +70,17 @@ object FirPropertyInitializationAnalyzer : AbstractFirPropertyInitializationChec
             symbol: FirPropertySymbol,
             node: VariableAssignmentNode
         ): Boolean {
+            if (symbol.fir.isVal && node.fir in capturedWrites) {
+                if (symbol.fir.isLocal) {
+                    reporter.reportOn(node.fir.lValue.source, FirErrors.CAPTURED_VAL_INITIALIZATION, symbol, context)
+                } else {
+                    reporter.reportOn(node.fir.lValue.source, FirErrors.CAPTURED_MEMBER_VAL_INITIALIZATION, symbol, context)
```

Comment:
Please note that this checker isn't invoked against member properties yet, and thus we don't see any changes in this commit/PR. Extending this to member properties in general is on my TODO list.